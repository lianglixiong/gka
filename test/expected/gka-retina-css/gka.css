.gka-wrap {
    width: 50px;
    height: 50px;
}

.animation {
    width: 50px;
    height: 50px;
    background-image: url("./img/gkatest_1.png");
    background-size: 50px 50px;
    background-repeat: no-repeat;
    animation-name: keyframes-name;
    animation-duration: 0.32s;
    animation-delay: 0s;
    animation-iteration-count: infinite;
    animation-fill-mode: forwards;
    animation-timing-function: steps(1);
}

@keyframes keyframes-name {
    0% {
        width: 50px;
        height: 50px;
        background-image: url("./img/gkatest_1.png");
        background-size: 50px 50px;
    }
    12.50% {
        background-image: url("./img/gkatest_2.png");
    }
    25.00% {
        background-image: url("./img/gkatest_3.png");
    }
    37.50% {
        background-image: url("./img/gkatest_4.png");
    }
    50.00% {
        background-image: url("./img/gkatest_5.png");
    }
    62.50% {
        background-image: url("./img/gkatest_6.png");
    }
    75.00% {
        background-image: url("./img/gkatest_7.png");
    }
    87.50%,
    100% {
        background-image: url("./img/gkatest_8.png");
    }
}

@media (-webkit-min-device-pixel-ratio: 2),
(min-resolution: 192dpi) {
    .gka-wrap {
        width: 25px;
        height: 25px;
    }
    .animation {
        width: 25px;
        height: 25px;
        background-image: url("./img/2x/gkatest_1.png");
        background-size: 25px 25px;
        background-repeat: no-repeat;
        animation-name: keyframes-name;
        animation-duration: 0.32s;
        animation-delay: 0s;
        animation-iteration-count: infinite;
        animation-fill-mode: forwards;
        animation-timing-function: steps(1);
    }
    @keyframes keyframes-name {
        0% {
            width: 25px;
            height: 25px;
            background-image: url("./img/2x/gkatest_1.png");
            background-size: 25px 25px;
        }
        12.50% {
            background-image: url("./img/2x/gkatest_2.png");
        }
        25.00% {
            background-image: url("./img/2x/gkatest_3.png");
        }
        37.50% {
            background-image: url("./img/2x/gkatest_4.png");
        }
        50.00% {
            background-image: url("./img/2x/gkatest_5.png");
        }
        62.50% {
            background-image: url("./img/2x/gkatest_6.png");
        }
        75.00% {
            background-image: url("./img/2x/gkatest_7.png");
        }
        87.50%,
        100% {
            background-image: url("./img/2x/gkatest_8.png");
        }
    }
}