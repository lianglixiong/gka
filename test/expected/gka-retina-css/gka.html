<html>
<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width,initial-scale=1,maximum-scale=1,user-scalable=0,maximum-scale=1">
    
    <title>gka-preview</title>
<link href="./gka.css" rel="stylesheet" type="text/css">
</head>
<body>

    <div class="gka-wrap">
        <div id="gka"></div>
    </div>
    <script>
    function preloadImage(names, cb, prefix){
        window.gkaCache = window.gkaCache || [];
        var n = 0,img,imgs = {};
        names.forEach(function(name) {
            img = new Image();
            window.gkaCache.push(img);
            img.onload = (function(name, img) {
                return function() {
                    imgs[name] = img;
                    (++n === names.length) && cb && cb(imgs);
                }
            })(name, img);
            img.src = (prefix || '') + name;
        });
    }

    preloadImage(["gkatest_1.png","gkatest_2.png","gkatest_3.png","gkatest_4.png","gkatest_5.png","gkatest_6.png","gkatest_7.png","gkatest_8.png"], function() {
        document.getElementById('gka').className += " animation"
    }, "img/" + (window.devicePixelRatio >= 2? '2x/': ''))
    </script>
</body>
</html>